### ISSUE TITLE
model_func of LeastSquareRegressor does not seem consistent

### ISSUE THREAD

**Issue by RobertSchueler** [VFDannenberg/hrosailing#11]
Unpacking tws, twa in a model_func with a single input variable doesnt work:

```
import numpy as np

from hrosailing.processing.pipeline import PolarPipeline, CurveExtension
from hrosailing.processing.pipelinecomponents import \
    ArrayHandler, LeastSquareRegressor

def mf(x, *args):
    tws, twa = x
    return args[0]

extension = CurveExtension(
    regressor = LeastSquareRegressor(model_func=mf, init_vals = (0,))
)

pp = PolarPipeline(
    extension = extension,
    handler = ArrayHandler(),
)

pp(np.random.random((1000,3)))
```

Writing the model function in tws and twa also doesnt work:

```
import numpy as np

from hrosailing.processing.pipeline import PolarPipeline, CurveExtension
from hrosailing.processing.pipelinecomponents import \
    ArrayHandler, LeastSquareRegressor

def mf(tws, twa, *args):
    return args[0]

extension = CurveExtension(
    regressor = LeastSquareRegressor(model_func=mf, init_vals = (0,))
)

pp = PolarPipeline(
    extension = extension,
    handler = ArrayHandler(),
)

pp(np.random.random((1000,3)))
```

**Comment by VFDannenberg** [comment]
In the first example: array unpacking doesn't work the way it is presented, for example

```python
x = np.array([[1, 2], [3, 4]])
a, b = x
```
would lead to

```python
a == np.array([1, 2])
b == np.array([3, 4])
```

The second example works fine for me?

**Comment by RobertSchueler** [comment]
First example:

This is the desired behaviour. Sorry if I have been ambiguous. To my knowledge scipy.optimize.curve_fit interprets inputs as (k,M)-shaped arrays where k is the number of features and M is the number of records.

Second example:

Still does not work for me (IndexError: tuple index out of range). I use scipy 1.7.1.

Even the following code gives an IndexError:


```
import numpy as np

from hrosailing.processing.pipeline import PolarPipeline, CurveExtension
from hrosailing.processing.pipelinecomponents import \
    ArrayHandler, LeastSquareRegressor
from hrosailing.processing.modelfunctions import ws_s_s_dt_wa_gauss_comb


extension = CurveExtension(
    regressor = LeastSquareRegressor(model_func = ws_s_s_dt_wa_gauss_comb)
)

pp = PolarPipeline(
    extension = extension,
    handler = ArrayHandler(),
)

pp(np.random.random((1000,3)))
```

**Comment by VFDannenberg** [comment]
fixed with commit 142f24866bdd4e6f037819db028412440e5a30ee
